examples/soap-calculator/test/helpers.ts

```diff
@@ -0,0 +1,31 @@
+import {CalculatorDSDataSource} from '../src/datasources/calculator-ds.datasource';
+import {CalculatorServiceProvider} from '../src/services/calculator.service';
+
+// tslint:disable-next-line:no-any
+export async function givenAConnectedDataSource(): Promise<any> {
+  const calculatorDataSource = new CalculatorDSDataSource();
+
+  return new Promise(async (resolve, reject) => {
+    try {
+      await calculatorDataSource.connect();
+      resolve(calculatorDataSource);
+    } catch (err) {
+      reject(err);
+    }
+  });
+}
+
+// tslint:disable-next-line:no-any
+export async function givenACalculatorService(): Promise<any> {
+  return new Promise(async (resolve, reject) => {
```

Comment:
Similarly here - no need to wrap the code in `new Promise`, just `return calculatorService` and let the async function take care of error propagation for you.